dependencies.qmd

```diff
@@ -8,7 +8,15 @@ In the Epiverse-TRACE community, we recognize the importance of carefully select
 
 We prefer to minimize dependencies unless they offer significant performance or usability benefits. This strategy helps us keep our Epiverse-TRACE packages lightweight and reduces the potential for compatibility issues. We prioritize well-established packages with a strong track record of staying on CRAN to limit accumulation of technical debt. For example, we adhere to [rOpenSci's guidance on recommended scaffolding](https://devguide.ropensci.org/building.html#recommended-scaffolding) and avoid unmaintained packages.
 
-In particular, we exercise caution when considering dependencies that rely on external libraries. Such dependencies can pose challenges [on shared computing platforms where users may lack super-user privileges](https://twitter.com/groundwalkergmb/status/1350336844022521857), or [when using webR](https://www.tidyverse.org/blog/2023/08/webr-0-2-0/#webassembly-system-libraries-for-r-packages). While these dependencies may offer valuable functionality, we carefully consider alternative solutions or consult with platform administrators to address potential issues and ensure seamless deployment.
+In particular, packages that are not available on CRAN or Bioconductor should not be included as dependencies. Only CRAN/Bioconductor packages that already went through a couple of releases and/or have more than 100 monthly users can be considered as dependencies.
```

Comment:
I agree with @TimTaylor 
I only have one question about packages actively maintained: What about mature packages? By "mature" I mean that they are stable enough so they do not need very frequent maintainance and, although they may have not been modified in a while, they still work in newer versions of R.

Are there any packages that may be considered "mature" according to that definition? Should we advise developers for or against using that kind of packages?

Reply:
Thanks, all good points!

By including a specific number of downloads, I had hoped to make the policy more actionable but I am now convinced it doesn't make much sense and it's more likely to be confusing than anything.

For "actively maintained" vs "mature packages", I think it's addressed in a quite balanced manner in the previous paragraph with the phrases "with a strong track record of staying on CRAN" / "avoid unmaintained packages". I think it acknowledges that some packages require less regular updates but it ensures that these packages stay up-to-date with the latest CRAN & package ecosystem evolutions. What do you think?

> Are there any packages that may be considered "mature" according to that definition?

Yes, I think there are such packages. R6 or here could be examples. But it's more likely to be observed in small packages as large ones are likely to be hit by breaking changes in R, a system library, etc. at some point and have even a patch release.
I would be curious to see if I can find an example with the last release more than 5 years ago.

Reply:
rematch may be a good example: https://github.com/gaborcsardi/rematch

Only one minor update since 2016.

In the same vein, rex has been going through a very slow maintenance: https://github.com/r-lib/rex

Common themes:

- built on rock solid foundation (perl regular expressions)
- at a quick glance, no serious issue reported on GitHub
- built and maintained by active & trusted members of the community